### ISSUE TITLE
Darien Gap is Open Again

### ISSUE THREAD

**Issue by kevinkreiser** [valhalla/valhalla#3250]
![image](https://user-images.githubusercontent.com/697548/128381976-ae9333bf-ad1b-401f-94ba-41a199561e98.png)

The good news is that our fixes for precision in binning fixed all the errant tiles that made everything connected. The bad news is that our band aide for the Darien Gap no longer works. We have 2 options:

1. expand the band aide. we have code that marks specific tiles as not neighbors in the connectivity map and cuts off connectivity there. right now there is only one pair of neighbors so we'd need to add more to keep the separation
2. implement the finer grained map that uses the bins rather than the tiles. on the surface this sounds great because its bow way more accurate and should remedy the darien gap (and other falsely seemingly connected areas), however this will come at the cost of start up time as one cannot simply see what tiles exist in the tileset, one must also see what cells in each tiles bins have entries in them. now this information is all available in each tiles header and so it only means reading like 500 bytes of each tile, but there are generally about 200k tiles in a global tileset. so we have about 100megabytes of random-ish (sequential but not contiguous) reads. in tile_extract mode this will probably be insignificant since the file is already open. in tile_dir mode we have to open every file once..

I'd prefer to implement `2` because its more correct but without an experiment on performance its hard to make that call. if it takes 5 minutes to start up on a rotational harddrive in tile_dir mode then its probably a feature killer. maybe connectivity map has to have two modes, coarse and fine in the config, rather than just on/off.

**Comment by dnesbitt61** [comment]
could an image file (ppm or otherwise) be created with the data and read in at runtime? This would avoid having to compute the (more complex) connectivity at startup. I guess the image would be a bit larger than before if you use bins rather than tiles, but would it be prohibitively large?

**Comment by kevinkreiser** [comment]
@dnesbitt61 true, we could require it to be precomputed rather than at start up, reminds me of the precomputed tileset that we are doing over in the swapping tilesets issue with @nilsnolde (cant find the issue right now)

**Comment by dnesbitt61** [comment]
You maybe wouldn't need to require it. Use it if the file exists (config options to set the path perhaps), otherwise compute it at start if the use_connectivity configuration option is true.

**Comment by kevinkreiser** [comment]
Yep that's what I assumed too just poorly worded

**Comment by dnesbitt61** [comment]
I did some timing and constructing loki with connectivity map for the planet (with memory mapped tar file) seems to take ~150ms per loki instance. Most of this time is spent forming the connectivity map (getting a list of tiles from the tar file seems to take ~30ms).
